hackathon_site/hardware/api_filters.py

```diff
@@ -0,0 +1,24 @@
+from django_filters import rest_framework as filters
+from .models import Hardware
+from .serializers import HardwareSerializer
+from django_filters.fields import Lookup
+
+
+class HardwareFilter(filters.FilterSet):
+    queryset = Hardware
+    serializer_class = HardwareSerializer
+
+    class Meta:
+        model = Hardware
+        fields = ["name", "quantity_available"]
+
+    def comma_separate_filter(self, queryset, value):
+        value_list = value.split(",")
+        queryset = super().filter(queryset, Lookup(value_list, "in"))
```

Comment:
hmm I'm not sure this is going to work, since `filter` is a method on a `Filter` subclass, not a `FilterSet`. You could create your own custom filter class, based on `CharFilter`:

```
class CommaSeparatedFilter(filters.CharFilter):
    lookup_expr = "in"
    
    def filter(self, qs, value):
        if value:
            value = [x.strip() for x in value.split(",")]
        
        return super().filter(qs, value)
```

And then use that, passing in `field_name=id` when instantiating it

Reply:
See below, there's one built in